infrastructure-playbooks/rolling_update.yml

```diff
@@ -536,25 +536,67 @@
         - (ceph_versions.get('stdout', '{}')|from_json).get('osd', {}) | length == 1
         - ceph_versions_osd | string is search("ceph version 12")
 
-- name: upgrade ceph mdss cluster
-  vars:
-    upgrade_ceph_packages: True
-  hosts: "{{ mds_group_name|default('mdss') }}"
-  serial: 1
-  become: True
+
+- name: upgrade ceph mdss cluster, deactivate all rank > 0
+  hosts: "{{ groups[mon_group_name|default('mons')][0] }}"
+  become: true
   tasks:
-    - name: stop ceph mds
+    - import_role:
+        name: ceph-defaults
+    - import_role:
+        name: ceph-facts
+
+    - name: set max_mds 1 on ceph fs
+      command: "{{ container_exec_cmd | default('') }} ceph --cluster {{ cluster }} fs set {{ cephfs }} max_mds 1"
+      changed_when: false
+
+    - name: wait until only rank 0 is up
+      command: "{{ container_exec_cmd | default('') }} ceph --cluster {{ cluster }} fs get {{ cephfs }} -f json"
+      changed_when: false
+      register: wait_rank_zero
+      retries: 200
+      delay: 1
+      until: (wait_rank_zero.stdout | from_json).mdsmap.in | length == 1 and (wait_rank_zero.stdout | from_json).mdsmap.in[0]  == 0
+
+    - name: get name of remaining active mds
+      command: "{{ container_exec_cmd | default('') }} ceph --cluster {{ cluster }} fs dump -f json"
+      changed_when: false
+      register: _mds_active_name
+
+    - name: set_fact mds_active_name
+      set_fact:
+        mds_active_name: "{{ [(_mds_active_name.stdout | from_json)['filesystems'][0]['mdsmap']['info'][item.key]['name']] }}"
+      with_dict: "{{ (_mds_active_name.stdout | from_json).filesystems[0]['mdsmap']['info'] }}"
+
+    - name: stop standby ceph mds
       systemd:
-        name: ceph-mds@{{ ansible_hostname }}
+        name: ceph-mds@{{ hostvars[item]['ansible_hostname'] }}
         state: stopped
         enabled: no
         masked: yes
-      when: not containerized_deployment | bool
+      delegate_to: "{{ item }}"
+      with_items: "{{ groups[mds_group_name] | difference(hostvars[groups[mon_group_name][0]]['mds_active_name']) }}"
```

Comment:
you don't need `hostvars[groups[mon_group_name][0]]` because we're already on the first monitor node